### PR TITLE
force events to be v03 in channel.

### PR DESCRIPTION
**Pull request by n3wscott**

### CODE REVIEW

File: pkg/pubsub/publisher/publisher.go [hunk 1]
```diff
@@ -62,6 +62,11 @@ func (a *Publisher) Start(ctx context.Context) error {
 }
 
 func (a *Publisher) receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
+	// Upgrade to supported transport version.
```
Comment (vaikas-google) [inline pkg/pubsub/publisher/publisher.go]: Just to make sure, this is totally lossless? If I put a v2 there, then it gets upgraded to v3, but my function expects v2, then this is fine and nothing is lost?

Comment (n3wscott) [reply pkg/pubsub/publisher/publisher.go]: pubsub does not support anything except for v03. The clients all should expect this

Comment (n3wscott) [reply pkg/pubsub/publisher/publisher.go]: the diff between v02 and v03 is minimal

**Pull request merged**